lib/read-body.js

```diff
@@ -0,0 +1,135 @@
+var pify = require('pify')
+var temp = require('fs-temp')
+var Busboy = require('busboy')
+
+var pump = pify(require('pump'))
+var onFinished = pify(require('on-finished'))
+
+var MulterError = require('./error')
+
+function drainStream (stream) {
+  stream.on('readable', stream.read.bind(stream))
+}
+
+function collectFields (busboy, limits) {
+  return new Promise(function (resolve, reject) {
+    var result = []
+
+    busboy.on('field', function (fieldname, value, fieldnameTruncated, valueTruncated) {
+      if (fieldnameTruncated) return reject(new MulterError('LIMIT_FIELD_KEY'))
+      if (valueTruncated) return reject(new MulterError('LIMIT_FIELD_VALUE', fieldname))
```

Comment:
When an error is thrown from `field` event, where are the uploaded files removed? Often files are parsed before fields.